main.go

```diff
@@ -6,12 +6,33 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 // Package gosh implements gosh command.
+//
+// To embed Gosh into your Go program use subpackages:
 //  * https://godoc.org/gosh-lang.org/gosh/tokens
 //  * https://godoc.org/gosh-lang.org/gosh/scanner
 //  * https://godoc.org/gosh-lang.org/gosh/ast
 //  * https://godoc.org/gosh-lang.org/gosh/parser
 //  * https://godoc.org/gosh-lang.org/gosh/objects
 //  * https://godoc.org/gosh-lang.org/gosh/interpreter
+//
+// Small example:
+//	code := `println("Hello, world!")`
+//
+//	scanner, err := scanner.New(code, nil)
+//	if err != nil {
+//		log.Fatal(err)
+//	}
+//
+//	parser := parser.New(scanner, nil)
+//	program := parser.ParseProgram()
+//	if parser.Errors() != nil {
+//		log.Fatal(parser.Errors())
+//	}
+//
+//	scope := objects.NewScope(objects.Builtin(os.Stdout))
+//	interpreter := interpreter.New(nil)
```

Comment:
```interpreter := interpreter.New(nil)```

Why did you decide to have the name for both variable and package?
When you use `interpreter` next time it might be a little bit unclear if you call library's function or this var's method.

Reply:
How you would call it?

Reply:
Well, to be honest your first implementation with `s`, `p` and `i` made more sense for me :)
But I don't insist.

Reply:
PTAL